RecoEgamma/ElectronIdentification/src/ElectronDNNEstimator.cc

```diff
@@ -94,6 +114,31 @@ std::map<std::string, float> ElectronDNNEstimator::getInputsVars(const reco::Gsf
   variables["hadronicOverEm"] = ele.hcalOverEcalValid() ? ele.hadronicOverEm() : 0;
   variables["full5x5_e2x5Max"] = ele.full5x5_e2x5Max();
   variables["full5x5_e5x5"] = ele.full5x5_e5x5();
+
+  variables["full5x5_sigmaIphiIphi"] = ele.full5x5_sigmaIphiIphi();
+  variables["1_minus_full5x5_e1x5_ratio_full5x5_e5x5"] = 1 - ele.full5x5_e1x5() / ele.full5x5_e5x5();
+  variables["full5x5_e1x5_ratio_full5x5_e5x5"] = ele.full5x5_e1x5() / ele.full5x5_e5x5();
+  variables["full5x5_r9"] = ele.full5x5_r9();
+  variables["gsfTrack.trackerLayersWithMeasurement"] =
+      (validKF) ? myTrackRef->hitPattern().trackerLayersWithMeasurement() : -1.;
```

Comment:
is it justified to name the variable as `gsfTrack.trackerLayersWithMeasurement` while the `trackerLayersWithMeasurement()` is being extracted from the `hitPattern()` of electron's `closestCtfTrackRef()` ?

Reply:
same comment for [missing hits](https://github.com/cms-sw/cmssw/blob/10c3665dbfe18fafc0c897ad5e96c627a7612774/RecoEgamma/ElectronIdentification/src/ElectronDNNEstimator.cc#L103-L104) and [chi2](https://github.com/cms-sw/cmssw/blob/10c3665dbfe18fafc0c897ad5e96c627a7612774/RecoEgamma/ElectronIdentification/src/ElectronDNNEstimator.cc#L108). The name says gsf, but they are extracted from kf track.

Reply:
@swagata87 You are right! It looks like ["closestCtfTrackNormChi2"](https://github.com/cms-sw/cmssw/blob/10c3665dbfe18fafc0c897ad5e96c627a7612774/RecoEgamma/ElectronIdentification/src/ElectronDNNEstimator.cc#L101) and ["gsfTrack.normalizedChi2"](https://github.com/cms-sw/cmssw/blob/10c3665dbfe18fafc0c897ad5e96c627a7612774/RecoEgamma/ElectronIdentification/src/ElectronDNNEstimator.cc#L108) are basically referring to the same kf track Chi2. 

@valsdav All the gsfTrack variables need to be extracted from the gsf track:
`"gsfTrack.normalizedChi2", "gsfTrack.missing_inner_hits", "gsfTrack.trackerLayersWithMeasurement"`
[Here](https://github.com/cms-sw/cmssw/blob/10c3665dbfe18fafc0c897ad5e96c627a7612774/RecoEgamma/ElectronIdentification/src/ElectronDNNEstimator.cc#L89) we can also add the gsftrack reference `ele.gsfTrack()` and then get all the gsf variables from this reference.

Reply:
ok! Let me change that. I just pushed the rename, but I will overwrite that commit.

Reply:
I moved the reference to the gsfTrack.